### ISSUE TITLE
Defense Bot Ammo Start-of-Round spawning issue

### ISSUE THREAD

**Issue by mrjpjr** [Regalis11/Barotrauma#10551]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

At the start of every round an error "Failed to spawn the initial item 'defensebotammobox' in the inventory of 'Defensebot' pops up.

### Reproduction steps

1) Craft a Defense Bot
2) Start a new round

Following popup will appear:
![bot ammo error_2_8](https://user-images.githubusercontent.com/58775950/204054981-b26d77cc-48d2-47c7-b1c2-c8a473e39bb6.jpg)

### Bug prevalence

Happens every time I play

### Version

0.20.8.0 (Unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@856d014. Happened because we tried to spawn the initial items in the inventory, which was full. The initial items should only be spawned when the character is created, so this commit should also fix potential issues where extra items appears inside character inventories after loading a saved game. I don't think we had such inventory definitions for friendly characters, but it could happen with custom pets.

On testing:
1. start a campaign
2. spawn a defensebot (console command is ok)
3. pet it (not sure if it makes difference)
4. save the game (e.g. by accomplishing a round)
5. load the save
